department-of-veterans-affairs/caseflow
# APPEALS-30890

Resolves [APPEALS-30890](https://jira.devops.va.gov/browse/APPEALS-30890)

# Description
Adds UI for selecting the type of status report
## Acceptance Criteria
- [ ] When selecting ‘Status’ report, Radio button selection between:
    - [ ] ‘All statuses’
        - [ ] Radio button should default to this option.
        - [ ] If user selects ‘All statuses’, all statuses will be included in report generation query.
    - [ ] ‘Specific status’
        - [ ] If user selects ‘Specific status’, Status selection check box options appear.
        - [ ] Multi-select Check box options for Status include: 
            - [ ] Incomplete
            - [ ] Completed
        - [ ] Statuses selected will be included in report generation query. 
- [ ] ‘Select type of status report’ Radio button selection between:
    - [ ] ‘Last action taken’
        - [ ] Radio button should default to this option.
        - [ ] If user selects ‘Last action taken’, the most recent activity row associated to a claim will be included on report.
    - [ ] ‘Summary’
        - [ ] If user selects ‘Summary’, each activity row associated to a claim will be included on report. 
- [ ] If user selects 'Specific status' and selects any variation of check boxes, then selects 'All statuses', then reselects 'Specific status' the check box selection will be cleared to no selection.
- [ ] Form validation: 
    - [ ] At least one checkbox selected if "Status" -> Specific Status is selected 



## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. Log in as ACBAUERVVHAH and go to `/decision_reviews/vha/report` 
3. Select Status for type of report
4. Verify options appear:
    - All Statuses
    - Specific Status
    - Under Select type of status report
      - Last Action Taken
      - Summary
5. Select Specific Status and verify the multi-select checkbox appears with Incomplete, In Progress, Complete
6. Try to submit the form, an error should show
7. Select some of the checkboxes, and then change to 'All Status' and back to 'Specific Status' to make sure the checkboxes clear


# Frontend
## User Facing Changes

<img width="935" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/9c51637c-4c12-4646-bc51-fa7e01a33ac7">
<img width="1121" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/c5e6c975-60ba-4005-a301-6beae83e649e">
<img width="1095" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/87ad6585-2c17-458a-ae35-d1c42a567545">


## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
